### PR TITLE
General Improvments

### PR DESCRIPTION
**Pull request by bartimaeusnek**
+ Added GT_Recipe_Remover Class and moved all removeRecipeByOutput from GT_CraftingRecipeLoader class to it
+ Pulled Alloysmelter recipes from CoreMod minetweaker script to the GT_MachineRecipeLoader
+ Fixed broken AlloySmelter Recipes for ItemCasings (take in 3 instead of 2 and give out 2 instead of 3, titanium uses plates instead of ingots)
+ added methodes to get the coremods blocks as ItemStacks
+ added a method to get the coremods items with a custom amount as stack
+ enhanched perfomance/ram usage a bit (by using final, byte instead of int and ++i instead of i++ in for loops)

CoreMod.zs needs to be edited, I could outcomment or delete the AlloySmelter Recipes

### CODE REVIEW

File: src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java [hunk 1]
```diff
@@ -1460,14 +1460,38 @@ private void run2() {
 
         // --- Stargates
         if (Loader.isModLoaded("GalaxySpace") && Loader.isModLoaded("SGCraft"))
-            for (int i = 0; i < 6; i++)
+            for (byte i = 0; i < 6; ++i)
```
Comment (bartimaeusnek) [inline src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java]: i++ :
    create a temporary copy of i
    increment i
    return the temporary copy

++i :
    increment i
    return i

File: src/main/java/com/dreammaster/gthandler/GT_Recipe_Remover.java [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+        GT_ModHandler.removeRecipeByOutput(GT_ModHandler.getModItem("opensecurity", "opensecurity.rfidReaderCard", 1L, 0), true, false, true);
+        GT_ModHandler.removeRecipeByOutput(GT_ModHandler.getModItem("opensecurity", "opensecurity.secureNetworkCard", 1L, 0), true, false, true);
+        GT_ModHandler.removeRecipeByOutput(GT_ModHandler.getModItem("opensecurity", "opensecurity.securityDoor", 1L, 0), true, false, true);
+        GT_ModHandler.removeRecipeByOutput(GT_ModHandler.getModItem("opensecurity", "opensecurity.securityDoorPrivate", 1L, 0), true, false, true);
+        GT_ModHandler.removeRecipeByOutput(GT_ModHandler.getModItem("opensecurity", "opensecurity.damageUpgrade", 1L, 0), true, false, true);
+        GT_ModHandler.removeRecipeByOutput(GT_ModHandler.getModItem("opensecurity", "opensecurity.cooldownUpgrade", 1L, 0), true, false, true);
+        GT_ModHandler.removeRecipeByOutput(GT_ModHandler.getModItem("opensecurity", "opensecurity.energyUpgrade", 1L, 0), true, false, true);
+        GT_ModHandler.removeRecipeByOutput(GT_ModHandler.getModItem("opensecurity", "opensecurity.movementUpgrade", 1L, 0), true, false, true);
+        //Computronics
```
Comment (Dream-Master) [inline src/main/java/com/dreammaster/gthandler/GT_Recipe_Remover.java]: Did the change make computronics recipe remove working @bartimaeusnek ?

**Comment by draknyte1** [comment]
> enhanched perfomance/ram usage a bit (by using final, byte instead of int and ++i instead of i++ in for loops)


Probably pointless optimisations. JVM probably does either or both already. 

I++ vs ++I have totally different use factors. As seen by your change, that wouldn't work unless I becomes -1.
I++ will give you the value of I, then increment the variable. 
++I will increment the variable and then return it.

File: src/main/java/com/dreammaster/gthandler/GT_Recipe_Remover.java (hunk 2 above)
Comment (bartimaeusnek) [reply src/main/java/com/dreammaster/gthandler/GT_Recipe_Remover.java]: havent touched computronics here @Dream-Master

File: src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java (hunk 1 above)
Comment (codewarrior0) [reply src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java]: Bytecode generated is identical in both cases!

**Comment by bartimaeusnek** [comment]
@draknyte1 in for-loops prefix and postfix do the same thing. This might be pointless in terms of performance gain, but it forces the compiler to use the faster implementation of incementaion here.

File: src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java (hunk 1 above)
Comment (bartimaeusnek) [reply src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java]: Then this is pointless.

**Comment by draknyte1** [comment]
No, because you'll get a different result for Object[i] if you use ++i vs i++ in a for loop. Hence why you'd have to start the index at -1, not 0, because ++i will skip 0.

**Comment by bartimaeusnek** [comment]
The way for loop is processed is as follows

1 First, initialization is performed (i=0)
2 the check is performed (i < n)
3 the code in the loop is executed.
4 the value is incremented (++0 = write 0+1, 0++ = read 0, write +1, in theory. in most cases the compiler will use ++i but sometimes it does not get it right.)
5 Repeat steps 2 - 4

This is the reason why, there is no difference between i++ and ++i in the for loop which has been used.

**Comment by Namikon** [comment]
I love such kind of conversations where you dissect the bytecode of something simple like this. It pulls your attention to stuff that can be learned where you didn't expect it 😁 
So far looks fine; The getItemStack methods for Blocks and Items probably should go in YAMCore. However, the ItemDescriptor class there is/was supposed to allow easy conversion between Items, Itemstacks with and without NBT data. But doesn't hurt to have these things here 👍

**Pull request merged**